src/brogue/Items.c

```diff
@@ -6033,6 +5982,31 @@ void relabel(item *theItem) {
     }
 }
 
+// If the most recently equipped item caused another item to be unequiped, is
+// uncursed, and both haven't left the inventory since, swap them back.
+void swapLastEquipment() {
+    item *theItem;
+    unsigned char command[10];
+
+    if (rogue.swappedIn == NULL || rogue.swappedOut == NULL) {
+        confirmMessages();
+        message("You have nothing to swap.", 0);
+        return;
+    }
+
+    equipItem(rogue.swappedOut, false, rogue.swappedIn);
```

Comment:
So if I understand right, equipItem may fail, but this doesn't appear to be handled here

Reply:
..good catch (I think).  It's reasonable for this to fail if we're trying to swap away from a cursed item.

The actual cursed messaging and not-reequipping is now handled by `equipItem`, but we should probably avoid recording the command.

Reply:
Ok, if `equipItem` fails we now bail out without recording input or ending the turn.